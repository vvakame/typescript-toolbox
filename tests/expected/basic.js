var Sample = (function () {
    function Sample() {
    }
    Sample.prototype.test = function (word) {
        if (typeof word === "undefined") { word = "world"; }
        return "Hello," + word;
    };
    return Sample;
})();

var s = new Sample();
s.test("TypeScript");
