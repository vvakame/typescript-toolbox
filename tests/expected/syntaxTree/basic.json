{
  "pos": 0,
  "end": 101,
  "flags": 0,
  "filename": "tmp.ts",
  "text": "class Sample{\ntest(word=\"world\"){return \"Hello,\"+word;}\n}\n\nvar s=new Sample();\ns.test(\"TypeScript\");\n",
  "syntacticErrors": [],
  "semanticErrors": [],
  "referencedFiles": [],
  "amdDependencies": [],
  "statements": [
    {
      "pos": 0,
      "end": 57,
      "flags": 0,
      "name": {
        "pos": 5,
        "end": 12,
        "text": "Sample"
      },
      "members": [
        {
          "pos": 13,
          "end": 55,
          "flags": 0,
          "name": {
            "pos": 13,
            "end": 18,
            "text": "test"
          },
          "parameters": [
            {
              "pos": 19,
              "end": 31,
              "flags": 0,
              "name": {
                "pos": 19,
                "end": 23,
                "text": "word"
              },
              "initializer": {
                "pos": 24,
                "end": 31,
                "text": "world"
              }
            }
          ],
          "body": {
            "pos": 32,
            "end": 55,
            "statements": [
              {
                "pos": 33,
                "end": 54,
                "expression": {
                  "pos": 39,
                  "end": 53,
                  "left": {
                    "pos": 39,
                    "end": 48,
                    "text": "Hello,"
                  },
                  "operator": 32,
                  "right": {
                    "pos": 49,
                    "end": 53,
                    "text": "word"
                  }
                }
              }
            ],
            "kind": 186
          }
        }
      ]
    },
    {
      "pos": 57,
      "end": 78,
      "declarations": [
        {
          "pos": 62,
          "end": 77,
          "flags": 0,
          "name": {
            "pos": 62,
            "end": 64,
            "text": "s"
          },
          "initializer": {
            "pos": 65,
            "end": 77,
            "func": {
              "pos": 68,
              "end": 75,
              "text": "Sample"
            },
            "arguments": []
          }
        }
      ]
    },
    {
      "pos": 78,
      "end": 100,
      "expression": {
        "pos": 78,
        "end": 99,
        "func": {
          "pos": 78,
          "end": 85,
          "left": {
            "pos": 78,
            "end": 80,
            "text": "s"
          },
          "right": {
            "pos": 81,
            "end": 85,
            "text": "test"
          }
        },
        "arguments": [
          {
            "pos": 86,
            "end": 98,
            "text": "TypeScript"
          }
        ]
      }
    }
  ],
  "nodeCount": 23,
  "identifierCount": 8,
  "version": "0",
  "isOpen": false,
  "languageVersion": 2,
  "identifiers": {
    "Sample": "Sample",
    "test": "test",
    "word": "word",
    "s": "s"
  }
}